include/kframework/lemmas/lemmas.k

```diff
@@ -28,14 +28,18 @@ module LEMMAS [symbolic]
     rule         chop(_V) <Int pow256 => true        [simplification]
 
     rule X *Int Y <Int pow256 => true requires Y <=Int maxUInt256 /Int X [simplification]
-    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification] 
+    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification]
 
   // ########################
   // Set Reasoning
   // ########################
 
-    rule X in (SetItem(Y) _   ) => true      requires X  ==Int Y [simplification]
-    rule X in (SetItem(Y) REST) => X in REST requires X =/=Int Y [simplification]
+    rule X in SetItem ( Y ) S:Set => ( X ==K Y andBool notBool ( X in S ) ) orBool ( X =/=K Y andBool X in S ) [simplification]
+    rule X in S:Set SetItem ( Y ) => ( X ==K Y andBool notBool ( X in S ) ) orBool ( X =/=K Y andBool X in S ) [simplification]
```

Comment:
The version of this rule that we have in `gateway-lemmas.k` is
```
rule X in SetItem ( Y ) S:Set => ( X ==Int Y ) orBool ( X in S )
  [simplification]
```
which seems better because we might have cases where `X ==K Y` but we don't know if `X in S` or not.

Reply:
By construction of sets, I believe that we have to have the stronger one, otherwise the set is malformed. This is similar to maps, where `(K -> _) M` means that `M` does not have `K` as a key. Am I correct, @ehildenb?